examples/example.hs

```diff
@@ -68,7 +68,9 @@ testHint =
       say $ show $ not bool_val
       emptyLine
       say "Here we evaluate an expression of type string, that when evaluated (again) leads to a string"
-      res <- interpret "head $ map show [\"Worked!\", \"Didn't work\"]" infer >>= flip interpret infer
+      res <- do
+        s <- interpret "head $ map show [\"Worked!\", \"Didn't work\"]" infer
+        interpret s infer
```

Comment:
You made a lot of code more pointfree; this is of course a question of style, but I think the pointfree style is more appropriate when using pure higher-order functions than when writing imperative code in `do` blocks. I have thus reverted the places where you followed hlint's suggestions to make things more pointfree. I have also found quite a few places which were also too pointfree for my taste prior to your changes, like here, so I'm making those pointful as well.